src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -0,0 +1,229 @@
+            return;
+        }
+
+        Promise.all([MyDataHelps.getDataCollectionSettings(), MyDataHelps.getExternalAccounts(), MyDataHelps.getDeviceInfo()])
+            .then(([settings, accounts, deviceInfo]) => {
+                setSettings(settings);
+                setDeviceExternalAccounts(accounts);
+                setPlatform(deviceInfo.platform);
+                setLoading(false);
+            });
+    }
+
+    useEffect(() => {
+        initialize();
+        MyDataHelps.on("applicationDidBecomeVisible", initialize);
+        return () => {
+            MyDataHelps.off("applicationDidBecomeVisible", initialize);
+        }
+    }, []);
+
+    if (loading) {
+        return null;
+    }
+
+    //Omron excluded by default
+    let accountTypes = props.accountTypes || ["Fitbit", "Garmin", "AppleHealth", "GoogleFit"];
+    if (!settings?.fitbitEnabled) {
+        accountTypes = accountTypes.filter(a => a != "Fitbit");
+    }
+    if (!settings?.garminEnabled) {
+        accountTypes = accountTypes.filter(a => a != "Garmin");
+    }
+    if (!settings?.queryableDeviceDataTypes.find(a => a.namespace == "AppleHealth")) {
+        accountTypes = accountTypes.filter(a => a != "AppleHealth");
+    }
+    if (!settings?.queryableDeviceDataTypes.find(a => a.namespace == "GoogleFit")) {
+        accountTypes = accountTypes.filter(a => a != "GoogleFit");
+    }
+    if(!accountTypes.length){
+        return null;
+    }
+
+    function getFitbitMenuItem() {
+        if (!accountTypes.includes("Fitbit")) {
+            return null;
+        }
+        return getExternalAccountMenuItem("Fitbit", getFitbitProviderID(), <img src={FitbitIcon} />);
+    }
+
+    function getGarminMenuItem() {
+        if (!accountTypes.includes("Garmin")) {
+            return null;
+        }
+        return getExternalAccountMenuItem("Garmin", getGarminProviderID(), <img src={GarminIcon} />);
+    }
+
+    function getOmronMenuItem() {
+        if (!accountTypes.includes("Omron")) {
+            return null;
+        }
+        return getExternalAccountMenuItem("Omron", getOmronProviderID(), <img style={{ borderRadius: "2px" }} src={OmronIcon} />);
+    }
+
+    function getExternalAccountMenuItem(providerName: string, providerID: number, icon: ReactElement) {
+        let externalAccount = deviceExternalAccounts?.find(a => a.provider.id == providerID);
+
+        let indicator = <div className="mdhui-connect-devices-menu-connect">{language("connect")}</div>;
+        let action: (() => void) | undefined = () => {
+            MyDataHelps.connectExternalAccount(providerID)
+        };
+
+        if (externalAccount) {
+            if (externalAccount.status == "fetchComplete" || externalAccount.status == "fetchingData") {
+                indicator = <div className="mdhui-connect-devices-menu-connected">{language("connected")}</div>;
+                action = undefined;
+            }
+            if (externalAccount.status == "fetchingData") {
```

Comment:
`indicator` and `action` are set twice when the status is `fetchingData`.

Reply:
Fixed